src/main/java/org/privacyidea/PrivacyIDEA.java

```diff
@@ -3,24 +3,50 @@
 import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
 import com.google.gson.JsonSyntaxException;
 
 import java.util.*;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+import static org.privacyidea.PIConstants.*;
+import static org.privacyidea.PIConstants.ASSERTIONCLIENTEXTENSIONS;
+import static org.privacyidea.PIConstants.AUTHENTICATORDATA;
+import static org.privacyidea.PIConstants.CLIENTDATA;
+import static org.privacyidea.PIConstants.CREDENTIALID;
+import static org.privacyidea.PIConstants.SIGNATUREDATA;
+import static org.privacyidea.PIConstants.USERHANDLE;
+import static org.privacyidea.PIResponse.getString;
+
+/**
+ * Copyright 2021 NetKnights GmbH - nils.behlen@netknights.it
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 public class PrivacyIDEA {
 
-    private final Configuration configuration;
-    private final PILoggerBridge log;
-    private final PISimpleLogBridge simpleLog;
+    private final PIConfig configuration;
+    private final IPILogger log;
+    private final IPISimpleLogger simpleLog;
     private final AtomicBoolean runPoll = new AtomicBoolean(true);
     private final Endpoint endpoint;
 
-    private PrivacyIDEA(Configuration configuration, PILoggerBridge logger, PISimpleLogBridge simpleLog) {
+    private PrivacyIDEA(PIConfig configuration, IPILogger logger, IPISimpleLogger simpleLog) {
```

Comment:
Its because the PrivacyIDEA class uses a builder pattern to be constructed like this:
https://github.com/privacyidea/sdk-java/blob/678bebf5bf728ca85a1d4e10fddc0c6a24087ac1/src/test/java/org/privacyidea/TestOTP.java#L31

It allowes to hide additional configuration possibilities for which sensible defaults exsist. It requires only the configuration params that are necessary to work at all (currently url and user agent).